src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.CoreCLR.cs

```diff
@@ -136,15 +137,21 @@ public static int OffsetToStringData
         [MethodImpl(MethodImplOptions.InternalCall)]
         public static extern bool TryEnsureSufficientExecutionStack();
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern object GetUninitializedObjectInternal(
-            // This API doesn't call any constructors, but the type needs to be seen as constructed.
-            // A type is seen as constructed if a constructor is kept.
-            // This obviously won't cover a type with no constructor. Reference types with no
-            // constructor are an academic problem. Valuetypes with no constructors are a problem,
-            // but IL Linker currently treats them as always implicitly boxed.
-            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)]
-            Type type);
+        private static unsafe object GetUninitializedObjectInternal(Type type)
+        {
```

Comment:
Undo the split between shared and runtime-specific implementation now that this has managed body?

The split makes it hard to see what's going on and it also makes things slower than what they need to be.

Reply:
There are still presumably three implementations of this method: one for coreclr, one for mono, one for corert. Right now the common entry point performs precondition checks then delegates to the runtime-specific implementation.

Is your suggestion that we remove the common entry point and have each implementation perform their own precondition checks?

Reply:
Yes, I think it would look better. The shared precondition checks are 4 lines. We are not sharing much.